### ISSUE TITLE
v0.3.6 is broken on firefox(-kde)v49 in chakraOS

### ISSUE THREAD

**Issue by ahjolinna** [i-rinat/freshplayerplugin#346]
chakraOS just updated the [freshplayerplugin to v0.3.6](https://chakralinux.org/code/gtk.git/tree/freshplayerplugin?id=37920f662f982a1f11b9aab5963ed48ce076b333) and now firefox won't recognize it, only by making a new profile it will find it again...BUT it still doesn't work aka. cant play videos...

backtrace logs will arrive soon

**Comment by ahjolinna** [comment]
basic firefox gdb output: (cutdown version)
```
.......
kf5.kio.core: Refilling KProtocolInfoFactory cache in the hope to find "wss"
[New Thread 0x7fff9a5fd700 (LWP 1737)]
kf5.kio.core: Refilling KProtocolInfoFactory cache in the hope to find "wss"
[New Thread 0x7fffa46fd700 (LWP 2585)]
[fresh 24350] not implemented: PPB_BrokerTrusted;0.3
[fresh] [error] [NP] caught Xlib error 8
[fresh] [error] [NP] caught Xlib error 8
[fresh] [error] [NP] caught Xlib error 171
[fresh] [error] [NP] caught Xlib error 171
[fresh] [error] ppb_graphics3d_create, glXMakeCurrent failed
[fresh] [error] ppb_graphics3d_create, glXMakeCurrent failed
[New Thread 0x7fff9adfe700 (LWP 2588)]
console.error: gnotifier: 
  Message: [Exception... "Component returned failure code: 0x80520006 (NS_ERROR_FILE_TARGET_DOES_NOT_EXIST) [nsIFile.directoryEntries]"  nsresult: "0x80520006 (NS_ERROR_FILE_TARGET_DOES_NOT_EXIST)"  location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> resource://jid1-oonoa6xbjznvlq-at-jetpack/lib/main.js :: deleteTempFiles :: line 252"  data: no]
  Stack:
    deleteTempFiles@resource://gre/modules/commonjs/toolkit/loader.js -> resource://jid1-oonoa6xbjznvlq-at-jetpack/lib/main.js:252:7
exports.onUnload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://jid1-oonoa6xbjznvlq-at-jetpack/lib/main.js:322:3
unload/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:58:7
unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:56:3
onunload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/unload.js:77:5
Observer<.observe@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/system/events.js:81:7
unload@resource://gre/modules/commonjs/toolkit/loader.js:795:3
unload/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/bootstrap.js:153:9
unload@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/bootstrap.js:149:12
shutdown@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/bootstrap.js:146:12
this.XPIProvider.callBootstrapMethod@resource://gre/modules/addons/XPIProvider.jsm:4785:9
this.XPIProvider.startup/<.observe@resource://gre/modules/addons/XPIProvider.jsm:2799:15
waitForSyncCallback@resource://services-common/async.js:98:7
makeSpinningCallback/callback.wait@resource://services-common/async.js:152:27
update@resource://gre/modules/services-sync/engines/addons.js:373:5
Store.prototype.applyIncoming@resource://services-sync/engines.js:341:7
applyIncoming@resource://gre/modules/services-sync/engines/addons.js:281:5
Store.prototype.applyIncomingBatch@resource://services-sync/engines.js:303:9
doApplyBatch@resource://services-sync/engines.js:1002:32
SyncEngine.prototype._processIncoming/newitems.recordHandler@resource://services-sync/engines.js:1128:9
set recordHandler/this._onProgress@resource://services-sync/record.js:623:9
Channel_onDataAvail@resource://services-sync/resource.js:539:7
waitForSyncCallback@resource://services-common/async.js:98:7
Res__request@resource://services-sync/resource.js:383:14
Res_get@resource://services-sync/resource.js:410:12
SyncEngine.prototype._processIncoming@resource://services-sync/engines.js:1135:18
SyncEngine.prototype._sync@resource://services-sync/engines.js:1555:7
WrappedNotify@resource://services-sync/util.js:165:21
Engine.prototype.sync@resource://services-sync/engines.js:670:5
_syncEngine@resource://services-sync/stages/enginesync.js:213:7
sync@resource://services-sync/stages/enginesync.js:163:15
onNotify@resource://gre/modules/services-sync/service.js:1278:7
WrappedNotify@resource://services-sync/util.js:165:21
WrappedLock@resource://services-sync/util.js:120:16
_lockedSync@resource://gre/modules/services-sync/service.js:1268:12
sync/<@resource://gre/modules/services-sync/service.js:1260:14
WrappedCatch@resource://services-sync/util.js:94:16
sync@resource://gre/modules/services-sync/service.js:1248:5

[New Thread 0x7fffa3efc700 (LWP 3608)]
[New Thread 0x7fff9d7ff700 (LWP 3609)]
1476105187911   Sync.ErrorHandler       ERROR   Sync was interrupted due to the application shutting down
[Thread 0x7fff9a5fd700 (LWP 1737) exited]
[Thread 0x7fffb93ff700 (LWP 23418) exited]
[Thread 0x7fffbadfd700 (LWP 23428) exited]
[Thread 0x7fffa46fd700 (LWP 2585) exited]
[Thread 0x7fffe0efe700 (LWP 23360) exited]
[Thread 0x7fffa68ff700 (LWP 23457) exited]
[New Thread 0x7fffe0efe700 (LWP 3611)]
[Thread 0x7fffe0efe700 (LWP 3611) exited]
[Thread 0x7fffd56ff700 (LWP 23404) exited]
[Thread 0x7fffcd5ff700 (LWP 23448) exited]
[New Thread 0x7fffd56ff700 (LWP 3612)]
[Thread 0x7fffd7bff700 (LWP 23374) exited]
[Thread 0x7fffd56ff700 (LWP 3612) exited]
[Thread 0x7fff9d7ff700 (LWP 3609) exited]
1476105188267   addons.xpi-utils        WARN    Error: XPI database modified after shutdown began (resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:458:17) JS Stack trace: this.XPIDatabase.saveChanges@XPIProviderUtils.js:458:17 < this.XPIDatabase.updateActiveAddons@XPIProviderUtils.js:1441:9 < this.XPIProvider.shutdown@XPIProvider.jsm:2854:7 < _startProvider/AMProviderShutdown/<@AddonManager.jsm:802:21 < Promise@Promise-backend.js:388:5 < _startProvider/AMProviderShutdown@AddonManager.jsm:800:16 < Barrier/this.client.addBlocker/promise</trigger@AsyncShutdown.jsm:720:23 < Barrier.prototype<._wait@AsyncShutdown.jsm:867:7 < Barrier.prototype<.wait@AsyncShutdown.jsm:851:28 < AddonManagerInternal.shutdownManager<@AddonManager.jsm:1199:15
1476105188269   addons.xpi-utils        WARN    Error: XPI database modified after shutdown began (resource://gre/modules/addons/XPIProvider.jsm -> resource://gre/modules/addons/XPIProviderUtils.js:458:17) JS Stack trace: this.XPIDatabase.saveChanges@XPIProviderUtils.js:458:17 < this.XPIDatabase.updateActiveAddons@XPIProviderUtils.js:1441:9 < this.XPIProvider.shutdown@XPIProvider.jsm:2854:7 < _startProvider/AMProviderShutdown/<@AddonManager.jsm:802:21 < Promise@Promise-backend.js:388:5 < _startProvider/AMProviderShutdown@AddonManager.jsm:800:16 < Barrier/this.client.addBlocker/promise</trigger@AsyncShutdown.jsm:720:23 < Barrier.prototype<._wait@AsyncShutdown.jsm:867:7 < Barrier.prototype<.wait@AsyncShutdown.jsm:851:28 < AddonManagerInternal.shutdownManager<@AddonManager.jsm:1199:15
[Thread 0x7fffbd174700 (LWP 23424) exited]

[Thread 0x7fffbb5fe700 (LWP 23419) exited]
[Thread 0x7fffe06ed700 (LWP 23372) exited]

```

**Comment by ahjolinna** [comment]
intesting I tried the new (offical) flashplugin v23.0.0.181 that is still in beta (but is the same what chrome uses) and it doesnt work either, I mean at least anymore with firefox49, it was fine a month ago ....could it be that its a problem with chakraOS firefox build? or somewhere

**Comment by ahjolinna** [comment]
okay it seems the problems has to do with some of my extensions/plugins... I have had a lot of problems with extensions/plugins after the ff49 introduced Electrolysis....

**Comment by i-rinat** [comment]
Tried to install Chakra OS in the virtual machine. And indeed, prebuilt freshplayerplugin wasn't working at all. I even tried to remove `pluginreg.dat` file multiple times, it didn't help. I then tried to download recent Firefox version from Mozilla archive. After removing `pluginreg.dat` it suddenly had Flash working. And after that Firefox from Chakra archives started working with Flash too. I even tried to remove whole Firefox profile directory with cache too. Still works.

Now when I try to recall all actions I performed, I think that I tried in-system Firefox first, it failed. Then I updated the whole OS. Tried again, but without removal of `pluginreg.dat`. Then was a delay of couple of days. And today I tried again. So maybe upgrade was the cause.

I saw similar issue with PepperFlash from Chrome OS before. It depended on a relatively new `libstdc++`, but it wasn't clear until some application tried to load the plugin. So that could happen in Chakra OS too.
